### ISSUE TITLE
Cannot Find Spreadsheet

### ISSUE THREAD

**Issue by StefanDuPreez** [aiguofer/gspread-pandas#33]
Hi,

I was using v1.2.2 and my script was connecting, reading and writing to my Google sheet as expected. After upgrading to v2.1.2 I am receiving 

> Traceback (most recent call last):
  File "C:/MyWork/Python/Backup/RPMSheetWriter/main.py", line 64, in <module>
    config=global_list.spread_config)
  File "C:\MyWork\Python\Backup\RPMSheetWriter\venv\lib\site-packages\gspread_pandas\spread.py", line 130, in __init__
    self.open(spread, sheet, create_sheet, create_spread)
  File "C:\MyWork\Python\Backup\RPMSheetWriter\venv\lib\site-packages\gspread_pandas\spread.py", line 199, in open
    self.open_spread(spread, create_spread)
  File "C:\MyWork\Python\Backup\RPMSheetWriter\venv\lib\site-packages\gspread_pandas\spread.py", line 249, in open_spread
    raise new_error
gspread.exceptions.SpreadsheetNotFound: Spreadsheet not found

When I use the `client.list_spreadsheet_files()` it does list my sheet, but if I try to read or write to it, I get the exception. 

Any help will be appreciated.

**Comment by aiguofer** [comment]
Hmm that's really strange. Are you using name, id, or URL in order to open the spreadsheet?

**Comment by StefanDuPreez** [comment]
Hi,

I am using the sheet id that is being returned from the `client.list_spreadsheet_files()`.

Just to add, if I use `client,.list_permissions()` and I pass in the Sheet id, I can see my service account does have write access to the sheet, so it is not a rights issue. If I downgrade to v1.2.2 the same code works perfectly.

Regards

**Comment by aiguofer** [comment]
Sorry for the delayed responses. I've been traveling and haven't had much time to check. That's really odd... And now I'm realizing this way of handling exceptions isn't great because I can't tell what exception you're actually getting. Do you feel comfortable with pdb?

If you could add:
```
        import pdb
        pdb.set_trace()
```
to line 229 (right before the try/except block) then we could make sure it's calling the right `open` function (should be `open_by_key` in this case) and see exactly what the exception is.

That being said, the code that opens the spreadsheet hasn't changed. My guess is that it's something related to the API permissions and/or scope for the project, but I don't remember changing those too much (although I did change the entire auth system in v2).

**Comment by StefanDuPreez** [comment]
Sorry for the delay, quite busy at the moment.

While debugging, I can see it trying to figure out what open method to call at line 223 of spread.py. 

It goes into the else and sets the following
`
open_func = self.client.open
` 
which is trying to use the title to open the sheet, so it is not using the id I am passing in.

The exception that occurs within the open function is the following https://imgur.com/a/Oi1kXj6

**Comment by aiguofer** [comment]
Interesting.... That's what I thought might be happening, but it's strange
because that code hasn't changed between the two mentioned versions.

How many characters is your ID, and does it contain any characters other
than `[a-zA-Z0-9-_]`?

**Comment by StefanDuPreez** [comment]
So I figured it out, the problem was between the keyboard and the chair.

In version 1.2.2 I initialized the Spread with:
`spread = Spread(email, id, config)`

That worked without problems, but in version 2.1.2 I had to change it to:
`spread = Spread(id, config)`

I compared the Spread initialize functions of the two versions with each other and I noticed there is quite a difference. I don't know why it took me so long to notice.

So I am sure it is safe to close this issue
